### PR TITLE
os/binfmt, framework/messaging: disable them by default

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. Application separation and loading is not a mandatory so that let's disable them by default.
2. Remove unnecessary conditionals, CONFIG_DEBUG and CONFIG_BINFMT_ENABLE.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cda5f7118dc0132743c3c837e3e51f8c06b26bf9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cda5f7118dc0132743c3c837e3e51f8c06b26bf9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2d378c0beeb2c4d05614a5b2def20a0e137f1bed] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2d378c0beeb2c4d05614a5b2def20a0e137f1bed] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [de952039a1e31c64e0fb6f7f6c092ac714ce9c2c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [de952039a1e31c64e0fb6f7f6c092ac714ce9c2c] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
Do not merge

File: build/configs/artik053/hello-protected/defconfig [hunk 1]
```diff
@@ -832,7 +832,7 @@ CONFIG_LIB_USRWORKSTACKSIZE=2048
 #
 # Binary Loader
 #
-# CONFIG_BINFMT_DISABLE is not set
+CONFIG_BINFMT_ENABLE=y
```
Comment (sijupunnnoose) [inline build/configs/artik053/hello-protected/defconfig]: Why do we enable binary loader in hello protected?

File: build/configs/artik053/net-protected/defconfig [hunk 2]
```diff
@@ -1071,7 +1071,7 @@ CONFIG_NETUTILS_JSON=y
 #
 # Binary Loader
 #
-# CONFIG_BINFMT_DISABLE is not set
+CONFIG_BINFMT_ENABLE=y
```
Comment (sijupunnnoose) [inline build/configs/artik053/net-protected/defconfig]: Ditto?

File: build/configs/imxrt1020-evk/tc_protected/defconfig [hunk 3]
```diff
@@ -899,7 +899,7 @@ CONFIG_C99_BOOL8=y
 #
 # Binary Loader
 #
-# CONFIG_BINFMT_DISABLE is not set
+CONFIG_BINFMT_ENABLE=y
```
Comment (sijupunnnoose) [inline build/configs/imxrt1020-evk/tc_protected/defconfig]: Ditto

File: build/configs/imxrt1050-evk/nxp_demo_protected/defconfig [hunk 4]
```diff
@@ -867,7 +867,7 @@ CONFIG_C99_BOOL8=y
 #
 # Binary Loader
 #
-# CONFIG_BINFMT_DISABLE is not set
+CONFIG_BINFMT_ENABLE=y
```
Comment (sijupunnnoose) [inline build/configs/imxrt1050-evk/nxp_demo_protected/defconfig]: Ditto

File: build/configs/imxrt1050-evk/tc_protected/defconfig [hunk 5]
```diff
@@ -902,7 +902,7 @@ CONFIG_C99_BOOL8=y
 #
 # Binary Loader
 #
-# CONFIG_BINFMT_DISABLE is not set
+CONFIG_BINFMT_ENABLE=y
```
Comment (sijupunnnoose) [inline build/configs/imxrt1050-evk/tc_protected/defconfig]: Ditto

File: build/configs/artik053/hello-protected/defconfig (hunk 1 above)
Comment (sunghan-chang) [reply build/configs/artik053/hello-protected/defconfig]: Right. That's why I left "do not merge". I will remove it on protected build when I am in office.

**Pull request closed without merging**